packages/block-editor/src/components/global-styles/border-panel.js

```diff
@@ -85,8 +85,10 @@ export default function BorderPanel( {
 	defaultControls = DEFAULT_CONTROLS,
 } ) {
 	const colors = useColorsPerOrigin( settings );
-	const decodeValue = ( rawValue ) =>
-		getValueFromVariable( { settings }, '', rawValue );
+	const decodeValue = useCallback(
+		( rawValue ) => getValueFromVariable( { settings }, '', rawValue ),
+		[ settings ]
+	);
```

Comment:
I wrapped it in `useCallback` to avoid the error below, but I'm a little unsure.

> The 'decodeValue' function makes the dependencies of useMemo Hook (at line 119) change on every render. To fix this, wrap the definition of 'decodeValue' in its own useCallback() Hook.

Reply:
The error makes sense given `decodeValue` was being defined on each render so wrapping it in a `useCallback` also seems reasonable. An alternative would be to move the definition of that function out of the component and pass it the settings too.

I think it's ok with the `useCallback` but don't have a strong opinion there. It also matches how `decodeColorValue` worked originally.